jyellick/plob/chaincode/assetregistry.go

```diff
@@ -177,11 +180,11 @@ func (ac *assetContext) create() ([]byte, error) {
 }
 
 func (ac *assetContext) lock() ([]byte, error) {
-	if ac.functionArg == nil {
-		return nil, fmt.Errorf("Must pass toChannel argument")
+	if !(ac.functionArgs != nil && (len(ac.functionArgs) == 2)) {
+		return nil, fmt.Errorf("Must pass toChaincode/toChannel arguments")
 	}
 
-	toChannel := string(ac.functionArg)
+	toChaincode, toChannel := string(ac.functionArgs[0]), string(ac.functionArgs[1])
```

Comment:
Indeed. I was torn between this and the "same-chaincode" approach (see: https://github.com/jyellick/hyperledger-fabric-alpha2-challenge/commit/a7a83c34d0244b5538428cda1cc3176dd790fca9#commitcomment-23506446). Will push a revision tomorrow morning.